src/app/blog/[slug]/components/SharePost.tsx

```diff
@@ -0,0 +1,46 @@
+'use client'
+
+import { useEffect, useState } from 'react'
+
+import { clsx } from 'clsx'
+
+import { Icon } from '@/components/Icon'
+import { touchTarget } from '@/components/Social'
+
+import { getSharePostSocialIcons } from '../utils/getSharePostSocialIcons'
+
+type SharePostProps = {
+  title: string
+  postTitle: string
+}
+
+export function SharePost({ title, postTitle }: SharePostProps) {
+  const [currentUrl, setCurrentUrl] = useState('')
+
+  useEffect(() => {
+    setCurrentUrl(window.location.href)
+  }, [])
+
+  const socialMediaIcons = getSharePostSocialIcons(currentUrl, postTitle)
```

Comment:
Could we hard-code the `BASE_URL` here instead?

Unless there's a reason behind having dynamic urls?


```suggestion
  const socialMediaIcons = getSharePostSocialIcons(BASE_URL, postTitle)
```